dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/ingest/IngestException.java

```diff
@@ -0,0 +1,48 @@
+package edu.harvard.iq.dataverse.ingest;
+
+import com.google.common.collect.Lists;
+import io.vavr.control.Option;
```

Comment:
Why not simply `java.util.Optional`?

Reply:
I kinda like `Option` api more, it has an option to do `ifPresentOrElse()` like Optional in java 9. It also has method like `getOrNull` for legacy (ofc you can also do it in Optional). So just preference.

Reply:
Mentioned features are not used at all in `IngestServiceBean`. Instead anti-pattern of `isDefined()` and `get()` are used. Pulling Vavr here is simply overkill. Let's keep it simple and stick to JDK.

Reply:
Yes, but if someone would want to use this api he would have to refactor the getter and all the method calls (although you can call it YAGNI, so I understand). Yes it is anti-pattern but in this case I thought it was more readable, your call.

Reply:
I'm not sure either `Option` or `Optional` are even needed here? You always call `get()` and then only pass the value to the setter of `IngestReport`. No logic is actually there and the field in `IngestReport` is nullable. Refactor `FileUtil#createIngestFailureReport(..)` as mentioned below and see if we can have a simple getter instead. Also only use `Optional` / `Option` if you mean to use it's API, if not, use regular null checks. Doing standard null checks using `Optional` / `Option` API, like it is now in `IngestServiceBean`, is much less readable than simple `if (x == null)` and not worth the trouble.

Reply:
Ok I have to agree in the 1st part. About the 2nd I do not exactly agree, for me the benefit besides Optional API is information. For the dev who never worked in some part of code `Optional` gives information that this field can be missing. You don't even need to look inside the class (I'm talking overall about Optional). There are also article mentioned by Stuart Mark: [article](https://blog.joda.org/2015/08). I know this is controversial since I've had conversations about this before and people are divided. But this is my opinion :P.

Reply:
Yes, `Optional` can be informative and useful, given proper context. If the only context it's usable is to convert it back into a nullable filed, than it's use is highly questionable. Also, my general preference is to use OOP encapsulation to avoid getters entirely wherever possible with all the problems they bring. But that's a story for another day ;)

Reply:
I have to agree in this case, kinda lost big picture here. Well that's the story I would like to hear ;)